File: Ekklesia.Spa/src/app/models/Member.ts
import { RoleEnum, RoleMapping } from './RoleEnum'

export class Member {
  id: string
  name: string
  phone: string
  photo: string
  roleName: string
  birthDay: string
  formFile?: File  
  BaseConverter: any

  private _role : RoleEnum;

  public get role() : RoleEnum {
    return this._role;
  }
  public set role(v : RoleEnum) {
    this._role = v;
    this.roleName = RoleMapping[v]
  }

  constructor() {
    this.id = '';
    this.name = '';
    this.phone = '';
    this.photo = '';
    this.roleName = '';
    this.birthDay = '';
    this._role = RoleEnum.INDEFINIDO;
  }   
  
  
  
  
  
}

File: Ekklesia.Entities/Entities/Member.cs
using Ekklesia.Entities.Enums;
using System;

namespace Ekklesia.Entities.Entities
{
    public class Member : BaseEntity
    {
        public string Name { get; set; }
        public string Phone { get; set; }
        public string Photo { get; set; }
        public Role Role { get; set; }
        public DateTime BirthDay { get; set; }

        public Member()
        {
            this.Name = string.Empty;
            this.Phone = string.Empty;
            this.Photo = string.Empty;
            this.Role = Role.INDEFINIDO;
        }
        public override bool Equals(object? obj)
        {
            Member? member = obj as Member;

            if (member == null)
            {
                return false;
            }
            return this.Id.Equals(member.Id);
        }

        public override int GetHashCode()
        {
            return this.Id.GetHashCode();
        }
    }
}
